src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,487 @@
+        {
+            get => _temperature;
+            set => SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn | FunctionSet.ExtendedMode : FunctionSet.PowerOff | FunctionSet.ExtendedMode), (byte)value });
+        }
+
+        #endregion
+
+        #region Primitive methods
+
+        /// <summary>
+        /// Refresh the screen
+        /// </summary>
+        public void Refresh() => SpiWrite(true, _byteMap);
+
+        /// <summary>
+        /// Clear the screen
+        /// </summary>
+        public void Clear()
+        {
+            for (int i = 0; i < _byteMap.Length; i++)
+            {
+                _byteMap[i] = 0;
+            }
+
+            SetCursorPosition(0, 0);
+            Refresh();
+        }
+
+        /// <summary>
+        /// Set the byte map
+        /// </summary>
+        /// <param name="byteMap">A 504 sized byte representing the full image</param>
+        public void SetByteMap(ReadOnlySpan<byte> byteMap)
+        {
+            if (byteMap.Length != ScreenSizeBytes)
+            {
+                throw new ArgumentException($"{nameof(byteMap)} length have to be {ScreenSizeBytes} bytes");
+            }
+
+            SetCursorPosition(0, 0);
+            byteMap.CopyTo(_byteMap);
+        }
+
+        #endregion
+
+        #region Text
+
+        /// <summary>
+        /// Write text
+        /// </summary>
+        /// <param name="text">The text to write</param>
+        public void Write(string text)
+        {
+            byte[] letter = new byte[6];
+            foreach (char c in text.ToCharArray())
+            {
+                // We only display specific characters and ignore the rest
+                // And only if it's in the screen
+                if (_position <= ScreenSizeBytes - 5)
+                {
+                    if (c >= 0x20 && c <= 0x7F)
+                    {
+                        for (int i = 0; i < 5; i++)
+                        {
+                            letter[i] = NokiaCharacters.Ascii[c - 0x20][i];
+                        }
+
+                        letter[5] = 0x00;
+                        SpiWrite(true, letter);
+                        _position += 5;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Write text and set cursor position to next line
+        /// </summary>
+        /// <param name="text">The text to write</param>
+        public void WriteLine(string text)
+        {
+            Write(text);
+            // calculate the position
+            int y = _position / Size.Width + 1;
+            y = y > Size.Height ? Size.Height : y;
+            SetCursorPosition(0, y);
+        }
+
+        /// <summary>
+        /// Write a raw byte stream to the display.
+        /// Used if character translation already took place.
+        /// </summary>
+        /// <param name="text">Text to print</param>
+        public void Write(ReadOnlySpan<byte> text)
```

Comment:
I'd refrain from calling this `text`, I suggest `buffer` instead.

Reply:
I just aligned on the interface for the LCD screens. so we should consider adjusting it